### ISSUE TITLE
[Feature request]: flatpak update should show versions installed and version to be updated to

### ISSUE THREAD

**Issue by rezad1393** [flatpak/flatpak#4515]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a feature request that matches the one I want to file, without success.

### Suggestion

[Feature request]: flatpak update should show versions installed and version to be updated to.
so that user can be informed.
right now just shows this
        ID                                     Branch               Op           Remote            Download
 1. [✓] com.obsproject.Studio.Locale           stable               u            flathub             1.5 kB / 47.0 kB
 2. [✓] org.gnome.Platform.Locale              40                   u            flathub             1.4 kB / 333.8 MB
 3. [✓] org.gnome.Platform                     40                   u            flathub             2.4 MB / 363.9 MB
 4. [✓] us.zoom.Zoom                           stable               u            flathub            54.8 MB / 54.9 MB
 
so I dont know what version zoom is going to be installed.
even though searching for zoom shows the numbered version in branch.

**Comment by mwleeds** [comment]
I think we should show the version for install operations too, and I suppose uninstall as well.

**Comment by Zalaxx** [comment]
This is one of the biggest missing feature of flatpak for me! I would like to be able to quickly know version what i'm updating my apps to, without needing to waste time going to the app's flathub page to find the current version and compare it to my installed version.
I would love to see this implemented soon!

ps: thanks for flatpak and flathub, it's really amazing

**Comment by rezad1393** [comment]
the weird thing is that flatpak update command shows this:
```

        ID                         Branch         Op         Remote          Download
 1.     org.signal.Signal          stable         u          flathub         < 168.5 MB

Proceed with these changes to the user installation? [Y/n]: 
```

but discover (kde software manager with flatpak integration) shows this with versions:
![Screenshot_20240530_132724](https://github.com/flatpak/flatpak/assets/7523164/aafce28b-3128-48b1-a96a-6bd98b00dae7)

**Comment by smcv** [comment]
> flatpak update command shows [no version numbers]
> but discover (kde software manager with flatpak integration) shows this with versions

Yes, this information exists internally (`libflatpak` can get the version number from AppStream metainfo), but the `flatpak(1)` command-line interface doesn't show it.

The command-line interface is somewhat limited in what it can display, because a character cell grid is just not a great user interface environment, particularly if we have to pessimistically assume that it's running in an 80x25 (or smaller!) terminal emulator.

The basic batch-mode CLI provided by Flatpak itself was never intended to be the only way to install Flatpak apps, and I would suggest that an interactive GUI, or an interactive TUI using a user interface toolkit of some sort (ncurses or similar, like Debian's `aptitude` frontend for `apt`), would make it much easier to display more information. But realistically, the maintainers of the core Flatpak project don't have the resources to be responsible for writing or maintaining that.

Anyone with better ideas about how the UI should look is welcome to implement a GUI or TUI around `libflatpak`, as both GNOME and KDE have already done.
